### DISCUSSION TITLE
Downloding hls video

### DISCUSSION THREAD

**Discussion by amit0021** [jhomlala/betterplayer#597] [General]
I looking to download m3u8 format (hls) video and play it on app when offline. Any refrence anyone have? Thanks for your time.

**Comment by jhomlala** [comment]
This is not a BetterPlayer issue. Please use Google/Stackoverflow: https://stackoverflow.com/questions/52765136/how-to-playm3u8-files-offline-in-vlc

**Comment by SinoMiles** [comment]
you can use this package to parse m3u8 content https://pub.dev/packages/flutter_hls_parser
and will get TS URL.
then sent a stream request and save the stream to the local file.
then you get so many TS files at your local document.
to use this FFmpeg convert all TS files to one mp4 https://pub.dev/packages/flutter_ffmpeg
